yarn/src/main/scala/org/apache/spark/deploy/yarn/ExecutorRunnable.scala

```diff
@@ -109,7 +110,14 @@ class ExecutorRunnable(
     }
 
     // Send the start request to the ContainerManager
-    nmClient.startContainer(container, ctx)
+    try {
+      nmClient.startContainer(container, ctx)
+    } catch {
+      case ex: YarnException =>
+        logError("Exception while start container %s on host %s:"
```

Comment:
Also, a couple nits: we should use String interpolation instead of `.format`, and "start" should be "starting".